rules/android_binary_internal/rule.bzl

```diff
@@ -29,22 +29,24 @@ def make_rule(
         attrs = ATTRS,
         implementation = impl,
         provides = _DEFAULT_PROVIDES,
-        additional_toolchains = []):
+        additional_toolchains = [],
+        additional_providers = []):
```

Comment:
Having the additional providers here is technically not needed. Happy to revert if it only makes sense supporting this at the android_library level for now.